cpp/src/arrow/util/basic_decimal.h

```diff
@@ -166,6 +167,317 @@ class ARROW_EXPORT GenericBasicDecimal {
   }
 };
 
+template <typename BaseType>
+class ARROW_EXPORT BasicDecimal {
```

Comment:
There's already a ton of "basic" types here, can you name it perhaps `SmallBasicDecimal`?

Reply:
renamed